watchconf-api/src/main/java/com/librato/watchconf/adapter/zookeeper/DynamicConfigZKAdapter.java

```diff
@@ -39,31 +37,18 @@ public DynamicConfigZKAdapter(final Class<T> clazz,
             }
         }
 
-        curatorFramework.sync().inBackground(new BackgroundCallback() {
-            @Override
-            public void processResult(CuratorFramework client, CuratorEvent event) throws Exception {
-                getAndSet(curatorFramework.getData().forPath(path));
-            }
-        }).forPath(path);
-
-        getAndSet(curatorFramework.getData().forPath(path));
-
         this.nodeCache = new NodeCache(curatorFramework, path);
         this.nodeCacheListener = new NodeCacheListener() {
             @Override
             public void nodeChanged() throws Exception {
-                curatorFramework.sync().inBackground(new BackgroundCallback() {
-                    @Override
-                    public void processResult(CuratorFramework client, CuratorEvent event) throws Exception {
-                        getAndSet(curatorFramework.getData().forPath(path));
-                        notifyListeners();
-                    }
-                }).forPath(path);
+                getAndSet(nodeCache.getCurrentData().getData());
+                notifyListeners();
             }
         };
 
         this.nodeCache.getListenable().addListener(nodeCacheListener);
         this.nodeCache.start(true);
+        getAndSet(nodeCache.getCurrentData().getData());
```

Comment:
Maybe to the point you made in the PR desc, but could this be done with nodeCache.rebuild() ?